src/pcp.rs

```diff
@@ -608,6 +610,7 @@ struct QueryShimGadget<F: FieldElement> {
 }
 
 impl<F: FieldElement> QueryShimGadget<F> {
+    #[allow(clippy::new_ret_no_self)]
```

Comment:
What's the clippy warning?

Reply:
It is unhappy because this returns `Box<dyn Gadget<F>>` instead of `Self` or `Box<Self>` or some other type that references `Self`. I'll see if I can fix this better by changing the function signature.

Reply:
I'd be happy with a signature change.